e2e/lookout_ingester_test/lookout_ingester_test.go

```diff
@@ -0,0 +1,200 @@
+	expiresAfter, _ := time.ParseDuration("1m")
+
+	defaultWaits.PollWait = pollWait
+	defaultWaits.ExpiresAfter = expiresAfter
+}
+
+func TestLookoutIngesterUpdatesPostgresWithJobInfo(t *testing.T) {
+	err := client.WithConnection(connectionDetails(), func(connection *grpc.ClientConn) error {
+		submitClient := api.NewSubmitClient(connection)
+		jobRequest := createJobRequest("personal-anonymous")
+
+		createQueue(submitClient, jobRequest, t)
+
+		submitResponse, err := client.SubmitJobs(submitClient, jobRequest)
+		assert.NoError(t, err)
+		assert.Equal(t, len(submitResponse.JobResponseItems), 1)
+
+		jobId := submitResponse.JobResponseItems[0].JobId
+		errStr := submitResponse.JobResponseItems[0].Error
+		assert.Empty(t, errStr)
+		assert.NotEmpty(t, jobId)
+
+		db, err := openPgDbTestConnection()
+		assert.NoError(t, err)
+
+		pJobRun, err := awaitJobRunEntry(db, jobId, defaultWaits.PollWait, defaultWaits.ExpiresAfter)
+		assert.NoError(t, err)
+		assert.NotNil(t, pJobRun)
+		assert.Equal(t, pJobRun.JobId, submitResponse.JobResponseItems[0].JobId)
+
+		pJob, err := awaitJobEntry(db, jobId, defaultWaits.PollWait, defaultWaits.ExpiresAfter, repository.JobSucceededOrdinal)
+		assert.NoError(t, err)
+		assert.NotNil(t, pJob)
+		assert.Equal(t, pJob.JobId, submitResponse.JobResponseItems[0].JobId)
+
+		return nil
+	})
+	assert.NoError(t, err)
+}
+
+func TestLookoutIngesterUpdatesPostgresWithJobInfoFailedJob(t *testing.T) {
+	err := client.WithConnection(connectionDetails(), func(connection *grpc.ClientConn) error {
+		submitClient := api.NewSubmitClient(connection)
+		jobRequest := createJobRequest("personal-anonymous")
+		jobRequest.JobRequestItems[0].PodSpec.Containers[0].Image = "https://wrongimagename/"
+
+		createQueue(submitClient, jobRequest, t)
+
+		submitResponse, err := client.SubmitJobs(submitClient, jobRequest)
+		assert.NoError(t, err)
+		assert.Equal(t, len(submitResponse.JobResponseItems), 1)
+
+		jobId := submitResponse.JobResponseItems[0].JobId
+		errStr := submitResponse.JobResponseItems[0].Error
+		assert.Empty(t, errStr)
+		assert.NotEmpty(t, jobId)
+
+		db, err := openPgDbTestConnection()
+		assert.NoError(t, err)
+
+		pJobRun, err := awaitJobRunEntry(db, jobId, defaultWaits.PollWait, defaultWaits.ExpiresAfter)
+		assert.NoError(t, err)
+		assert.NotNil(t, pJobRun)
+		assert.Equal(t, pJobRun.JobId, submitResponse.JobResponseItems[0].JobId)
+		assert.Equal(t, pJobRun.Succeeded, false)
+
+		pJob, err := awaitJobEntry(db, jobId, defaultWaits.PollWait, defaultWaits.ExpiresAfter, repository.JobFailedOrdinal)
+		assert.NoError(t, err)
+		assert.NotNil(t, pJob)
+		assert.Equal(t, pJob.JobId, submitResponse.JobResponseItems[0].JobId)
+		assert.Equal(t, repository.JobFailedOrdinal, pJob.State)
+
+		return nil
+	})
+	assert.NoError(t, err)
+}
+
+type PartialJob struct {
+	JobId string
+	State int
+}
+
+func awaitJobEntry(db *sql.DB, jobId string, pollWait time.Duration, expiresAfter time.Duration, expectedState int) (*PartialJob, error) {
```

Comment:
It's worth noting that in general this can only be guaranteed to work for terminal states.  For non-terminal states there are two cases where it can fail:
- The state is updated in the database twice (or more) between polls
- The state is conflated out by the lookout ingester and thus is never written to the db at all.
I think so far you've only used it for terminal states so all is good, but probably warrants a comment (and maybe an error if the user passes a non terminal state).

The same also goes for `awaitJobRunEntry`